### PR TITLE
Add SIG feature flag check to use-publicize-config

### PR DESCRIPTION
**Pull request by gmjuhasz**
This adds the check to the Initial State on the UI side for the SIG feature flag.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1203876111187910-as-1203969267014316

## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

Test this with its' corresponding wpcom change: D101773-code
Sandbox that patch, and check out this change. In `projects/plugins/social/src/js/editor.js` import in the `isSocialImageGeneratorEnabled` check and log check its value with and without the `publicize-social-image-generator` blog sticker.

### CODE REVIEW

File: projects/js-packages/publicize-components/changelog/add-social-image-generator-feature-flag
Comment (danielpost) [inline projects/js-packages/publicize-components/changelog/add-social-image-generator-feature-flag]: Do you think we can make this a bit more descriptive? I think it might end up in the changelog for Jetpack and Jetpack Social.

**Review by danielpost — COMMENTED** [review]
This looks fine to me! Just a minor comment about the changelog. Also, let's make sure we don't merge this until D101773-code is merged.

File: projects/plugins/jetpack/modules/sharedaddy/admin-sharing.js
Comment (danielpost) [inline projects/plugins/jetpack/modules/sharedaddy/admin-sharing.js]: These changes don't belong in this PR, right?

**Review by danielpost — APPROVED** [review]

File: projects/js-packages/publicize-components/src/hooks/use-publicize-config/index.js
Comment (pablinos) [inline projects/js-packages/publicize-components/src/hooks/use-publicize-config/index.js:97]: I feel like we could probably do something better to pluck out these values, both here and in the PHP. Something for a follow-up perhaps.

**Review by pablinos — APPROVED** [review]
Looks good! I can see the value changing as I add and remove the blog sticker.

File: projects/js-packages/publicize-components/src/hooks/use-publicize-config/index.js
Comment (gmjuhasz) [reply projects/js-packages/publicize-components/src/hooks/use-publicize-config/index.js:97]: Yeah, good idea. Should we create a task for that in our Inbox?

**Pull request merged**
